Make `ReflectionProperty#setAccessible()` and `ReflectionMethod#setAccessible()` no-op

Ocramius opened the pull request:
RFC: https://wiki.php.net/rfc/make-reflection-setaccessible-no-op

Follow-up to #5411: also removes the `ReflectionProperty#setAccessible(false)` and `ReflectionMethod#setAccessible(false)`.

Make `ReflectionProperty#setAccessible()` and `ReflectionMethod#setAccessible()` no-op

With this patch, it is no longer required to call `ReflectionProperty#setAccessible()` or
`ReflectionMethod#setAccessible()` at all: calling these methods will no longer have any
effect on the behavior of `ReflectionMethod` and `ReflectionProperty`.

If a userland consumer already got to the point of accessing object/class information via reflection,
it makes little sense for `ext/reflection` to disallow accessing `private`/`protected` symbols ever.

## TODOs

 * [x] squash
 * [x] add full RFC text to commit message

nikic commented:
I'd probably also add a deprecation if `setAccessible(false)` is called that tells you that it doesn't do anything. We can leave `setAccessible(true)` alone to avoid needlessly complicating cross version code.

Ocramius commented:
Similar to https://github.com/php/php-src/pull/5411#issuecomment-615761407, I would probably open an alternate PR to add the deprecation there.

Effectively 3 levels:

 1. change the default (only)
 2. make `setAccessible()` a dummy
 3. deprecate `setAccessible()` *and* make it a dummy

Ocramius commented:
Cleaned this up: will raise an RFC, but an explicit deprecation of `Reflection*#setAccessible()` will only be viable after https://wiki.php.net/rfc/deprecated_attribute is discussed and moved on.

Adding a runtime side-effect here is worse, and I'd just avoid that.

I'll mention this decision in the RFC.

beberlei commented:
@Ocramius imho the dependency on the attribute RFC is not really there, ad it would also trigger the runtime deprecation. Marking the method as deprecated is PHP project convention, so your preference against runtime deprecations is ranked lower here :)

Ocramius commented:
> so your preference against runtime deprecations is ranked lower here :)

Will roll with it anyway: runtime side-effects are more damaging anyway.

Again, let's discuss that on the RFC, if needed, but I don't see an immediate need for any deprecation to be raised either.

Ocramius reviewed (commented):
@patrickallaert or @ramsey can you take this to merge into `master` and into `8.1.x`?, according to voting results?

Ref: 

 * https://externals.io/message/115068#115351
 * https://marc.info/?l=php-internals&m=162566643315847&w=2
 * https://wiki.php.net/rfc/make-reflection-setaccessible-no-op

Waited for green build before pinging :)
